### PR TITLE
Deploy ako in bootstrap cluster with tanzu-framework

### PR DESCRIPTION
**Pull request by HanFa**
**What this PR does / why we need it**:
This PR migrates the responsibility of AKO deployment in the bootstrap cluster. Previously, the AKO is deployed by the add-on generated by AKO-operator. Now, we leverage the 03_customizations to directly deploy the resources for AKO.

**Which issue(s) this PR fixes**:

**Describe testing done for PR**:

**Special notes for your reviewer**:

**Release note**:

```release-note

```
**New PR Checklist**

- [x] Ensure PR contains only public links or terms
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Squash the commits in this branch before merge to preserve our git history
- [x] If this PR is just an idea or POC, use a [Draft PR](https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/about-pull-requests#draft-pull-requests) instead of a full PR
- [x] Add appropriate [kind label](../docs/release/kind-labels.md) according to what type of issue is being addressed.

### CODE REVIEW

File: pkg/v1/providers/ytt/03_customizations/02_avi/add-ako.yaml [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+--- #@ ako_serviceaccount()
+--- #@ ako_configmap()
+--- #@ ako_clusterrole()
+--- #@ ako_clusterrolebinding()
+--- #@ ako_statefulset()
+--- #@ ako_ingressclass()
+--- #@ hostrules_crd()
+--- #@ httprules_crd()
+--- #@ aviinfrasettings_crd()
+#@ end
```
Comment (XudongLiuHarold) [inline pkg/v1/providers/ytt/03_customizations/02_avi/add-ako.yaml:21]: new line

File: pkg/v1/providers/ytt/03_customizations/02_avi/add-ako.yaml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+#@ load("ako-deployment.lib.yaml", "ako_namespace", "ako_serviceaccount",
+#@    "ako_configmap", "ako_clusterrole", "ako_clusterrolebinding",
+#@    "ako_statefulset", "ako_ingressclass",
+#@    "hostrules_crd", "httprules_crd", "aviinfrasettings_crd")
+
+#! render the ako artifacts
+#@ if data.values.PROVIDER_TYPE == "vsphere" and data.values.TKG_CLUSTER_ROLE != "workload" and data.values.AVI_ENABLE and data.values.AVI_CONTROL_PLANE_HA_PROVIDER:
+--- #@ ako_namespace()
+--- #@ ako_serviceaccount()
```
Comment (XudongLiuHarold) [inline pkg/v1/providers/ytt/03_customizations/02_avi/add-ako.yaml:12]: you can combine service accont, cluster role and cluster role binding together as rbac to reduce function you are using.

File: pkg/v1/providers/ytt/03_customizations/02_avi/add_akoo.yaml [hunk 3]
```diff
@@ -2,15 +2,15 @@
...
+#@ load("ako-operator-deployment.lib.yaml", "akoo_deployment", "avi_credentials", "avi_certificate")
 
+#! render the ako-operator artifacts
 #@ if data.values.PROVIDER_TYPE == "vsphere" and data.values.TKG_CLUSTER_ROLE != "workload" and data.values.AVI_ENABLE and data.values.AVI_CONTROL_PLANE_HA_PROVIDER:
 --- #@ crd()
 --- #@ service()
 --- #@ role()
 --- #@ role_binding()
 --- #@ akoo_deployment()
---- #@ ako_deployment_config()
```
Comment (XudongLiuHarold) [inline pkg/v1/providers/ytt/03_customizations/02_avi/add_akoo.yaml:13]: Should also remove ako deployment config lib

File: pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml [hunk 4]
```diff
@@ -0,0 +1,580 @@
...
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: clusterName
+            - name: DISABLE_STATIC_ROUTE_SYNC
+              valueFrom:
+                configMapKeyRef:
+                  name: avi-k8s-config
+                  key: disableStaticRouteSync
+            - name: SUBNET_IP
```
Comment (XudongLiuHarold) [inline pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml:210]: Should aligin with AKO 1.5.2(1.5.3)'s template, you can refer here: https://github.com/vmware/load-balancer-and-ingress-services-for-kubernetes/tree/release-1.5.2/helm/ako/templates

File: pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml [hunk 5]
```diff
@@ -0,0 +1,580 @@
...
+metadata:
+  name: avi-k8s-config
+  namespace: #@ avi_system_namespace()
+data:
+  controllerIP: #@ data.values.AVI_CONTROLLER
+  serviceEngineGroupName: #@ data.values.AVI_SERVICE_ENGINE_GROUP
+  cloudName: #@ data.values.AVI_CLOUD_NAME
+  clusterName: #@ cluster_namespaced_name()
+  apiServerPort: 8080
+  subnetIP: #@ ip_and_prefix()[0]
```
Comment (XudongLiuHarold) [inline pkg/v1/providers/ytt/03_customizations/02_avi/ako-deployment.lib.yaml:81]: Should algin with AKO 1.5.2 or 1.5.3's template

**Review by XudongLiuHarold — COMMENTED** [review]
Also, need to test it e2ely before merge

File: pkg/v1/providers/ytt/03_customizations/02_avi/add_akoo.yaml (hunk 3 above)
Comment (HanFa) [reply pkg/v1/providers/ytt/03_customizations/02_avi/add_akoo.yaml:13]: Could you clarify? Do you mean we don't need the `akoo_deployment` `avi_credentials` and `avi_certificate` here?

File: pkg/v1/providers/ytt/03_customizations/02_avi/add-ako.yaml (hunk 1 above)
Comment (HanFa) [reply pkg/v1/providers/ytt/03_customizations/02_avi/add-ako.yaml:21]: yep

**Comment by HanFa** [comment]
Will recreate this PR since the fork is disconnected. Close this for now.

**Pull request closed without merging**